### ISSUE TITLE
Create tests that emit the right metadata for linters

### ISSUE THREAD

**Issue by mansona** [embroider-build/content-tag#19]
Look at this file and "import" the tests https://github.com/patricklx/ember-template-tag/blob/main/__tests__/parse-templates.test.ts#L28

**Comment by NullVoxPopuli** [comment]
This is required to unblock Glint: https://github.com/typed-ember/glint/pull/615

turns out half the tooling out there needs line/col information.

Example: https://github.com/typed-ember/glint/blob/main/packages/environment-ember-template-imports/-private/environment/preprocess.ts#L18

**Comment by ef4** [comment]
I would add that: it would be fine to do an exact port of the existing parse API that just emits location information. But I'm pretty sure every place that uses that API also immediately follows up by slicing and dicing the source to replace the content tags.

So it would be even better to 
 - make our `preprocess` accept an optional replacement callback that controls what we transpile the content tags into
 - make it emit relocation information alongside the transformed content

This would eliminate the need for other systems like glint or the eslint plugin to do their own source mangling after calling content-tag.

**Comment by patricklx** [comment]
I think ember template lint uses only location info.
Also, i do not think we need some callback transform.
It willl just complicate things. Since one needs to distinguish between class body and others.
We just need another format which can be parsed and does not change any location information around the templates.

**Comment by ef4** [comment]
> I think ember template lint uses only location info.

No, ember-template-lint does exactly the kind of slicing I'm talking about in [_applyFixes](https://github.com/ember-template-lint/ember-template-lint/blob/3abccb86c799971bbe8eee03c4b6298b5bbc324a/lib/linter.js#L221-L284). 

> It willl just complicate things. Since one needs to distinguish between class body and others.

But this is the whole point of why we made content-tag. It already knows that difference! The proper place to detect that difference is in the parser. A lot of the way the existing system worked is hacks to get around the fact that the previous parser can't tell the difference, and we can get rid of those hacks now.

**Comment by patricklx** [comment]
Oh, i understand it wrong then.
You mean slice original source to extract template content? That's what the meta data of ember-templatr-tag also emits.

2.
What do you mean by the replacement callback then? Maybe can you give an example, then i might understand it better.
